src/main/scala/nl/tradecloud/kafka/KafkaSubscriber.scala

```diff
@@ -62,20 +63,41 @@ class KafkaSubscriber(
       .groupedWithin(batchingSize, batchingInterval)
       .map(group => group.foldLeft(CommittableOffsetBatch.empty) { (batch, elem) => batch.updated(elem) })
       .mapAsync(parallelism = 3) { msg =>
-        log.debug("Committing offset, msg={}", msg)
+        log.debug("committing offset, msg={}", msg)
 
         msg.commitScaladsl().map { result =>
-          log.debug("Committed offset, msg={}", msg)
+          log.debug("committed offset, msg={}", msg)
           result
         }
       }
   }
 
-  def atLeastOnce(flow: Flow[KafkaMessage, CommittableOffset, _]): Future[Done] = {
+  val commitSink: Sink[CommittableOffset, _] = {
+    commitFlow.to(Sink.ignore)
```

Comment:
Could you explain how this indirection works?

Reply:
In order to use the directTo flow shape we need a commit Sink. This create a commit sink. We can ignore the result.